great_expectations/checkpoint/v1_checkpoint.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, List, Union, cast
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations import project_manager
+from great_expectations._docs_decorators import public_api
+from great_expectations.checkpoint.actions import ValidationAction  # noqa: TCH001
+from great_expectations.compatibility.pydantic import BaseModel, validator
+from great_expectations.core.serdes import _IdentifierBundle
+from great_expectations.core.validation_config import ValidationConfig
+from great_expectations.render.renderer.renderer import Renderer
+
+if TYPE_CHECKING:
+    from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+    from great_expectations.data_context.store.validation_config_store import (
+        ValidationConfigStore,
+    )
+
+
+class Checkpoint(BaseModel):
+    """
+    A Checkpoint is the primary means for validating data in a production deployment of Great Expectations.
+
+    Checkpoints provide a convenient abstraction for running a number of validations and triggering a set of actions
+    to be taken after the validation step.
+
+    Args:
+        validations: List of validation configs to be run.
+        actions: List of actions to be taken after the validations are run.
+
+    """
+
+    name: str
+    validations: List[ValidationConfig]
+    actions: List[ValidationAction]
+    id: Union[str, None] = None
+
+    class Config:
+        arbitrary_types_allowed = (
+            True  # Necessary for compatibility with ValidationAction's Marshmallow dep
+        )
+        """
+        When serialized, the validations field should be encoded as a set of identifiers.
```

Comment:
I don't understand the position of this comment between 2 attributes of `Config` (`arbitrary_types_allowed` and `json_encoders`). It sees like ti should either above `arbitrary_types_allowed` as a top level comment in `Config` or as part of the class comment below `Args`. It's not really a user facing comment though since they don't need t worry about this so maybe at the top of Config?